include/digitsDFA.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __DIGITSDFA_H__
+#define __DIGITSDFA_H__
+
+#include "dfa.h"
+#include <string>
+
+int error(char c); // used in error state
+int checkNonZero(char c); // check for non zero digits at start state
+int checkDigits(char c); // check for digits in accepting state
+
+class DigitsDFA : public Dfa {		
```

Comment:
I think ValidNumberDfa or something of that sorts is a better name than DigitsDFA. Makes it more clear what this DFA is trying to detect.